runner/src/main.rs

```diff
@@ -701,20 +716,28 @@ fn run_example(opt: &RunExamples, example: &Example) -> Step {
     Step::Multiple {
         name: example.name.to_string(),
         steps: vec![
-            vec![
-                Step::Multiple {
-                    name: "build wasm modules".to_string(),
-                    steps: application
-                        .modules
-                        .iter()
-                        .map(|(name, target)| build_wasm_module(name, target, &example.name))
-                        .collect(),
-                },
-                Step::Single {
-                    name: "build application".to_string(),
-                    command: build_application(&application),
-                },
-            ],
+            if example.applications.is_empty() {
+                vec![]
+            } else {
```

Comment:
Shouldn't there be a similar check here if application variant is provided but unsupported?